File: frontend_flutter/fruit_market/lib/models/ReturnRequest.dart
class ReturnRequest {
  final String returnId;
  final String orderId;
  final String userId;
  final String reason;
  final String? description;
  final String? imageUrl;
  final String status;
  final double refundAmount;
  final DateTime createdAt;
  final DateTime? approvedAt;
  final DateTime? completedAt;
  final DateTime? rejectedAt;
  final String? rejectReason;
  final String statusText;

  ReturnRequest({
    required this.returnId,
    required this.orderId,
    required this.userId,
    required this.reason,
    this.description,
    this.imageUrl,
    required this.status,
    required this.refundAmount,
    required this.createdAt,
    this.approvedAt,
    this.completedAt,
    this.rejectedAt,
    this.rejectReason,
    required this.statusText,
  });

  factory ReturnRequest.fromJson(Map<String, dynamic> json) {
    return ReturnRequest(
      returnId: json['returnId'] ?? '',
      orderId: json['orderId'] ?? '',
      userId: json['userId'] ?? '',
      reason: json['reason'] ?? '',
      description: json['description'],
      imageUrl: json['imageUrl'],
      status: json['status'] ?? 'pending',
      refundAmount: (json['refundAmount'] as num?)?.toDouble() ?? 0,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
      approvedAt: json['approvedAt'] != null
          ? DateTime.parse(json['approvedAt'])
          : null,
      completedAt: json['completedAt'] != null
          ? DateTime.parse(json['completedAt'])
          : null,
      rejectedAt: json['rejectedAt'] != null
          ? DateTime.parse(json['rejectedAt'])
          : null,
      rejectReason: json['rejectReason'],
      statusText: json['statusText'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'returnId': returnId,
      'orderId': orderId,
      'userId': userId,
      'reason': reason,
      'description': description,
      'imageUrl': imageUrl,
      'status': status,
      'refundAmount': refundAmount,
      'createdAt': createdAt.toIso8601String(),
      'approvedAt': approvedAt?.toIso8601String(),
      'completedAt': completedAt?.toIso8601String(),
      'rejectedAt': rejectedAt?.toIso8601String(),
      'rejectReason': rejectReason,
      'statusText': statusText,
    };
  }
}

File: backend_dotnet/fruit_api/fruit_api/Models/ReturnRequest.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("ReturnRequests")]
public class ReturnRequest
{
    [Key]
    [Column("returnId")]
    [StringLength(20)]
    public string ReturnId { get; set; } = null!;

    [Required]
    [Column("orderId")]
    public string OrderId { get; set; } = null!;

    [Required]
    [Column("userId")]
    public string UserId { get; set; } = null!;

    [Required]
    [MaxLength(255)]
    [Column("reason")]
    public string Reason { get; set; } = string.Empty;

    [Column("description")]
    public string? Description { get; set; }

    [MaxLength(500)]
    [Column("imageUrl")]
    public string? ImageUrl { get; set; }

    [Column("status")]
    [StringLength(30)]
    public string Status { get; set; } = "pending";

    [Column("refundAmount", TypeName = "decimal(12,2)")]
    public decimal RefundAmount { get; set; }

    [Column("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    [Column("approvedAt")]
    public DateTime? ApprovedAt { get; set; }

    [Column("completedAt")]
    public DateTime? CompletedAt { get; set; }

    [Column("rejectedAt")]
    public DateTime? RejectedAt { get; set; }

    [MaxLength(500)]
    [Column("rejectReason")]
    public string? RejectReason { get; set; }

    // Navigation properties
    [ForeignKey("OrderId")]
    public Order? Order { get; set; }

    [ForeignKey("UserId")]
    public User? User { get; set; }
}